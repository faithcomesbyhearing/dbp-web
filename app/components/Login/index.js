/**
 *
 * Login
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import SvgWrapper from '../SvgWrapper';
import FacebookAuthentication from '../../containers/FacebookAuthentication';
import GoogleAuthentication from '../../containers/GoogleAuthentication';

class Login extends React.PureComponent {
	constructor(props) {
		super(props);
		this.state = {
			password: '',
			email: '',
			signInActive: false,
			staySignedIn: false,
		};
	}

	componentDidUpdate(nextProps) {
		if (
			this.props.socialLoginLink &&
			nextProps.socialLoginLink !== this.props.socialLoginLink
		) {
			window.open(this.props.socialLoginLink, '_self');
		}
	}

	handlePasswordChange = (e) => {
		this.setState({ password: e.target.value });
		this.viewError();
	};

	handleEmailChange = (e) => {
		this.setState({ email: e.target.value });
		this.viewError();
	};

	handleSendingLogin = (e) => {
		e.preventDefault();
		this.props.sendLoginForm({
			email: this.state.email,
			password: this.state.password,
			stay: this.state.staySignedIn,
		});
	};

	handleStayLoggedInChange = (e) => {
		this.setState({
			staySignedIn: e.target.checked,
		});
	};

	toggleSignInForm = (state) => {
		this.setState({ signInActive: state });
		this.viewError();
	};

	get signInComponent() {
		const { errorMessageViewed, selectAccountOption, errorMessage } =
			this.props;

		return (
			<form onSubmit={this.handleSendingLogin}>
				<span className={'sign-in-input'}>
					<SvgWrapper
						className="icon"
						width="30px"
						height="30px"
						fill="#fff"
						svgid="e-mail"
					/>
					<input
						id={'email'}
						autoComplete={'email'}
						className={errorMessageViewed ? '' : 'error'}
						type={'email'}
						placeholder="E-mail"
						onChange={this.handleEmailChange}
						value={this.state.email}
					/>
				</span>
				<span className={'sign-in-input'}>
					<SvgWrapper
						className="icon"
						width="26px"
						height="26px"
						fill="#fff"
						svgid="lock"
					/>
					<input
						id={'password'}
						autoComplete={'current-password'}
						className={errorMessageViewed ? '' : 'error'}
						type="password"
						placeholder="Password"
						onChange={this.handlePasswordChange}
						value={this.state.password}
					/>
				</span>
				<div className="sign-in-button">
					<input
						className="login-checkbox"
						id={'login-checkbox'}
						type="checkbox"
						onChange={this.handleStayLoggedInChange}
					/>
					<label htmlFor={'login-checkbox'} className="text">
						Remember Me
					</label>
					<button type="submit" className="login-button">
						Sign In
					</button>
				</div>
				{!errorMessageViewed ? (
					<div className="login-error-message">
						<SvgWrapper className={'icon'} svgid={'warning'} />
						<span className={'error-text'}>
							{errorMessage
								? `${errorMessage} ( Click Below to Reset )`
								: 'Username or Password is incorrect. Please try again.'}
						</span>
					</div>
				) : null}
				<section className="forgot-password">
					<span
						role="button"
						id={'reset-password-button'}
						tabIndex={0}
						className="link"
						onClick={() => {
							selectAccountOption('password_reset');
							this.toggleSignInForm(false);
						}}
					>
						Reset your password.
					</span>
				</section>
			</form>
		);
	}

	viewError = () => {
		if (!this.props.errorMessageViewed) {
			this.props.viewErrorMessage();
		}
	};

	render() {
		const {
			socialLoginLink,
			socialMediaLogin,
			activeDriver,
			oauthError,
			readOauthError,
			oauthErrorMessage,
		} = this.props;

		return (
			<>
				{this.signInComponent}
				<FacebookAuthentication
					oauthError={oauthError}
					oauthErrorMessage={oauthErrorMessage}
					activeDriver={activeDriver}
					readOauthError={readOauthError}
					socialMediaLogin={socialMediaLogin}
					socialLoginLink={socialLoginLink}
				/>
				<GoogleAuthentication
					activeDriver={activeDriver}
					socialMediaLogin={socialMediaLogin}
					socialLoginLink={socialLoginLink}
				/>
				{/* <DonateButton /> */}
			</>
		);
	}
}

Login.propTypes = {
	sendLoginForm: PropTypes.func,
	socialMediaLogin: PropTypes.func,
	viewErrorMessage: PropTypes.func,
	selectAccountOption: PropTypes.func,
	readOauthError: PropTypes.func,
	socialLoginLink: PropTypes.string,
	oauthError: PropTypes.bool,
	oauthErrorMessage: PropTypes.string,
	errorMessage: PropTypes.string,
	activeDriver: PropTypes.string,
	errorMessageViewed: PropTypes.bool,
};

export default Login;
