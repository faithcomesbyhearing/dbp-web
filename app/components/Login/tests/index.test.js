import React from 'react';
import renderer from 'react-test-renderer';
import Login from '../index';

/* eslint-disable react/prop-types */
jest.mock('../../../containers/GoogleAuthentication', () => function googleAuthenticationMock(props) {
  return (
<div
		role={'button'}
		id={'google-login'}
		tabIndex={0}
		className={'google'}
		onClick={props.handleSocialLogin}
>
		Sign in with Google
</div>
);
});
jest.mock('../../../containers/FacebookAuthentication', () => function facebookAuthenticationMock(props) {
  return (
<div
		role={'button'}
		id={'facebook-login'}
		tabIndex={0}
		className={'facebook'}
		onClick={props.handleSocialLogin}
>
		Sign in with Facebook
</div>
);
});
/* eslint-enable react/prop-types */

const props = {
	sendLoginForm: jest.fn(),
	socialMediaLogin: jest.fn(),
	viewErrorMessage: jest.fn(),
	selectAccountOption: jest.fn(),
	readOauthError: jest.fn(),
	socialLoginLink: 'google',
	oauthErrorMessage: '',
	errorMessage: '',
	activeDriver: 'google',
	oauthError: false,
	errorMessageViewed: false,
};

describe('<Login /> component', () => {
	it('should match snapshot with expected props', () => {
		const tree = renderer.create(<Login {...props} />);
		expect(tree).toMatchSnapshot();
	});
});
