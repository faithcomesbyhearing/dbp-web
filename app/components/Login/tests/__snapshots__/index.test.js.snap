// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> component should match snapshot with expected props 1`] = `
[
  <form
    onSubmit={[Function]}
  >
    <span
      className="sign-in-input"
    >
      <svg
        className="icon"
        fill="#fff"
        height="30px"
        width="30px"
      >
        <use
          xlinkHref="/svglist.svg#e-mail"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
      <input
        autoComplete="email"
        className="error"
        id="email"
        onChange={[Function]}
        placeholder="E-mail"
        type="email"
        value=""
      />
    </span>
    <span
      className="sign-in-input"
    >
      <svg
        className="icon"
        fill="#fff"
        height="26px"
        width="26px"
      >
        <use
          xlinkHref="/svglist.svg#lock"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
      <input
        autoComplete="current-password"
        className="error"
        id="password"
        onChange={[Function]}
        placeholder="Password"
        type="password"
        value=""
      />
    </span>
    <div
      className="sign-in-button"
    >
      <input
        className="login-checkbox"
        id="login-checkbox"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="text"
        htmlFor="login-checkbox"
      >
        Remember Me
      </label>
      <button
        className="login-button"
        type="submit"
      >
        Sign In
      </button>
    </div>
    <div
      className="login-error-message"
    >
      <svg
        className="icon"
      >
        <use
          xlinkHref="/svglist.svg#warning"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
      <span
        className="error-text"
      >
        Username or Password is incorrect. Please try again.
      </span>
    </div>
    <section
      className="forgot-password"
    >
      <span
        className="link"
        id="reset-password-button"
        onClick={[Function]}
        role="button"
        tabIndex={0}
      >
        Reset your password.
      </span>
    </section>
  </form>,
  <div
    className="facebook"
    id="facebook-login"
    role="button"
    tabIndex={0}
  >
    Sign in with Facebook
  </div>,
  <div
    className="google"
    id="google-login"
    role="button"
    tabIndex={0}
  >
    Sign in with Google
  </div>,
]
`;
